mbed_devices/_internal/candidate_device.py

```diff
@@ -59,62 +58,29 @@ def __set__(self, instance: object, value: Sequence) -> None:
         if not value or not isinstance(value, (list, tuple)):
             raise FilesystemMountpointError(f"{self.name} must be set to a non-empty list or tuple.")
 
-        instance.__dict__[self.name] = value
+        instance.__dict__[self.name] = tuple(value)
```

Comment:
Not part of this PR, but I just realized that `Sequence` is not really a valid type here. There's a disconnect between the type hint and type check later on. String is a valid `Sequence` type, yet it's rejected via `isinstance` check.

Reply:
Good point, I'll tighten the typing up here.

Reply:
Done in 351d3ce